### PR TITLE
docs/decisions: split release process into multiple ADRs and expand them

### PR DESCRIPTION
**Pull request by jecluis**
Instead of having a single ADR for the release process, which includes the general architecture and methodology, as well a rather long (yet inaccurate) description of all the steps to perform a release, this patch set proposes splitting the release process in three different ADRs:

1. Release Methodology - which describes the what, with the pretty branch diagrams and what not
2. Release Steps - containing the rather verbose description of the release, step by step, followed by a rather more condensed version of commands to be executed to successfully perform a release
3. Release Testing - currently describing only one very basic testing scenario, but should be expanded to cover more ground

We also add a note on the existing Release Process ADR stating that document is superseded by the new ADRs.

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: docs/decisions/0007-release-process.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+and expanding them: [Release Methodology][adr_methodology],
+[Release Steps][adr_steps], and [Release Testing][adr_testing].
+
+[adr_methodology]: /docs/decisions/0015-release-methodology.md
+[adr_steps]: /docs/decisions/0016-release-steps.md
+[adr_testing]: /docs/decisions/0017-release-testing.md
+
+## Context and Problem Statement
 
 Releasing is an essential process for the `s3gw` process. Given the project is
```
Comment (votdev) [inline docs/decisions/0007-release-process.md]: Shouldn't it read like this?

```suggestion
Releasing is an essential process for the `s3gw` project.
```

File: docs/decisions/0015-release-methodology.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
+# Release Methodology for the s3gw project
+
+## Context and Problem Statement
+
+Releasing is an essential process for the `s3gw` project. Given the project is
+composed by various sub-projects, that need to be prepared, tested, and
+eventually released, the Release Process is not trivial.
```
Comment (votdev) [inline docs/decisions/0015-release-methodology.md:7]:
```suggestion
eventually released, the *Release Process* is not trivial.
```
or
```suggestion
eventually released, the [Release Process][process_adr] is not trivial.
```

File: docs/decisions/0017-release-testing.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+This document supersedes the [Release Process ADR][process_adr].
+
+## Note Before
+
+This document requires expansion. The whole team is encouraged to enhance this
+document with further testing scenarios.
+
+## Validate deployment via Helm Chart
+
+1. Ensure a kubernetes distribution with Longhorn and cert-manager installed is available. E.g.,
```
Comment (votdev) [inline docs/decisions/0017-release-testing.md]:
```suggestion
1. Ensure a Kubernetes distribution with Longhorn and cert-manager installed is available. E.g.,
```

**Review by votdev — CHANGES_REQUESTED** [review]

File: docs/decisions/0015-release-methodology.md (hunk 2 above)
Comment (jecluis) [reply docs/decisions/0015-release-methodology.md:7]: in this case, `Release Process` refers to the entirety of the process, and is a term. I don't think it should be emphasized, nor should it reference the document we are now considering outdated.

**Comment by jecluis** [comment]
@vmoutoussamy would you mind reviewing?

**Review by votdev — APPROVED** [review]
LGTM

**Review by vmoutoussamy — APPROVED** [review]
LGTM. I really like the split in 3 different "topics" and that there is a reference in each one to the others. Great job!

File: docs/decisions/0007-release-process.md [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+and expanding them: [Release Methodology][adr_methodology],
+[Release Steps][adr_steps], and [Release Testing][adr_testing].
+
+[adr_methodology]: /docs/decisions/0015-release-methodology.md
+[adr_steps]: /docs/decisions/0016-release-steps.md
+[adr_testing]: /docs/decisions/0017-release-testing.md
+
+## Context and Problem Statement
+
+Releasing is an essential process for the `s3gw` project. Given the project is
```
Comment (Ameenasuhani) [inline docs/decisions/0007-release-process.md:15]: https://github.com/aquarist-labs/s3gw/blob/c0e27c53fb42970b2c4125cd6748a639a926108d/docs/decisions/0007-release-process.md?plain=1#L22 can you please fix the hyperlink?

Comment (jecluis) [reply docs/decisions/0007-release-process.md:15]: Addressing that is outside the scope of this PR.

**Pull request merged**